javascript/javascript3/week3/ScreenShots/script.js

```diff
@@ -0,0 +1,254 @@
+ * @param {object} event - form submit.
+ */
+async function getScreenShot(event) {
+  event.preventDefault();
+  const targetSite = document.getElementById("target-address").value;
+
+  const buttonElement = document.getElementById("get-screenshot-button");
+
+  const options = {
+    method: "GET",
+    headers: {
+      "X-RapidAPI-Key": screenshotKEY,
+      "X-RapidAPI-Host": "website-screenshot6.p.rapidapi.com"
+    }
+  };
+
+  buttonElement.innerText = "Fetching your url...";
+
+  try {
+    const response = await fetch(
+      `${baseSCR_URL}?url=${targetSite}&width=1920&height=1080`,
+      options
+    );
+    if (!response.ok) {
+      throw new Error(response.message);
+    }
+    buttonElement.innerText = "Prepairing the picture for display...";
+    const result = await response.json();
+
+    const screenshotUrl = result.screenshotUrl;
+    document.getElementById("target-address").value = "";
+    const screeenshotToRender = await postScreenshot(
+      screenshotUrl,
+      targetSite,
+      currentUser
+    );
+
+    renderScreenshot(screeenshotToRender);
+    buttonElement.innerText = "Get your screenshot";
+  } catch (error) {
+    screenshotElement.innerText = `Cannot fetch your url, please check its validity, error: ${error.message}`;
+  }
+}
+
+/**
+ * Returns a hash code from a string - to emulate secure storage of passwords.
+ * @param  {String} str The string to hash.
+ * @return {Number}    A 32bit integer
+ * @see http://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/
+ * @returns {Number} - unique sequence of digits for a given string.*/
+function hashCode(str) {
+  let hash = 0;
+  for (let i = 0, len = str.length; i < len; i++) {
+    let chr = str.charCodeAt(i);
+    hash = (hash << 5) - hash + chr;
+    hash |= 0; // Convert to 32bit integer
+  }
+  return hash;
+}
+
+/**
+ * Saves the addres of the screenshot with metadata to crudcrud.com.
+ * @param {String} url - web address with the screenshot image.
+ * @param {String} target - web adress of the site, from which the screenshot was taken.
+ * @param {Object} user - metadata of the user, which saved the screenshot.
+ * @returns {Object} result - the object with all the added data (for rendering at a later stage).
+ */
+async function postScreenshot(url, target, user) {
+  const body = {
+    url,
+    target,
+    userId: user.userId
+  };
+  const response = await fetch(`${crudcrudURL}/${crudcrudKEY}/screenshots`, {
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json"
+    },
+    method: "POST",
+    body: JSON.stringify(body)
+  });
+
+  const result = await response.json();
+
+  return result;
+}
+
+/**
+ * Fetches screenshot data for the current user, then calls rendering function with the array of
+ * screnshot objects to render.
+ * @param {String} user - user id, currently implemented as user email.
+ */
+async function getScreenshots(user) {
```

Comment:
As there is only a single letter that differentiates this method from getScreenshot, you could possibly rename it to getAllScreenshots to avoid confusion.

Reply:
Great suggestion! I still remember my pain when I was trying to find some name for the function - And I just should have added "All". Will remember it for future , thanks once again for this!